### PR TITLE
SMO-473 Update Login screen UI according to Design

### PR DESCRIPTION
**Pull request by OlehPiskorskyj**
Simulator Screenshot - iPhone 14 Pro - 2023-09-13 at 08 40 30
Simulator Screenshot - iPhone 14 Pro - 2023-09-13 at 08 41 15

### CODE REVIEW

File: SenacorOnlineBanking/Components/SCButton/SCButton.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        }
+    }
+    
+    init(viewModel: SCButtonViewModel) {
+        self.viewModel = viewModel
+    }
+}
+
+extension SCButton {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCButton/SCButton.swift:30]: What is the reason to put this into extension?

File: SenacorOnlineBanking/Components/SCButton/SCButton.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    var contentView: some View {
+        Text(viewModel.title)
+            .font(Font.custom("Roboto-Bold", size: 16))
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCButton/SCButton.swift:33]: please create enum for font and sizes like `.buttonTitle`

File: SenacorOnlineBanking/Components/SCButton/SCButtonViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
+//
+//  SCButtonViewModel.swift
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 12/09/2023.
+//
+
+final class SCButtonViewModel {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCButton/SCButtonViewModel.swift:8]: Why not to use struct?

File: SenacorOnlineBanking/Components/SCButton/SCButtonViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    let title: LocalizedStrings
+    let foregroundColor: ColorAssets
+    let backgroundColor: ColorAssets
+    let disabled: Bool
+    var action: (() -> Void)?
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCButton/SCButtonViewModel.swift:13]: What is the reason to use `var`?

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+//
+//  Created by Oleh Piskorskyi on 12/09/2023.
+//
+
+import SwiftUI
+
+struct SCInputField: View {
+    
+    enum FocusedField {
+        case unSecure
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:13]: sugestion `case planText`

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        case secure
+    }
+    
+    @ObservedObject var viewModel: SCInputFieldViewModel
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:17]: private

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    @FocusState var focused: FocusedField?
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:18]: private

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            )
+            .onChange(
+                of: focused,
+                perform: { viewModel.onFocusChangeAction($0 != nil) }
+            )
+        }
+    }
+}
+
+extension SCInputField {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:49]: The same question

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    var secureField: some View {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:50]: at least private and please consider function e.g. `func makeSecureField()`

File: SenacorOnlineBanking/Components/SecureInfoView.swift [hunk 10]
```diff
@@ -0,0 +1,32 @@
+//
+//  SecureInfoView.swift
+//  SenacorOnlineBanking
+//
+//  Created by Oleh Piskorskyi on 13/09/2023.
+//
+
+import SwiftUI
+
+struct SecureInfoView: View {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SecureInfoView.swift:10]: Missing SC prefix?

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+        case secure
+    }
+    
+    @ObservedObject var viewModel: SCInputFieldViewModel
+    @FocusState var focused: FocusedField?
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            Text(viewModel.title)
+                .font(Font.custom("Roboto-Bold", size: 20))
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:23]: The same

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    }
+}
+
+extension SCInputField {
+    var secureField: some View {
+        HStack(spacing: 8) {
+            ZStack {
+                TextField(viewModel.hint?.localized ?? "", text: $viewModel.value)
+                    .focused($focused, equals: .unSecure)
+                    .font(Font.custom("Roboto-Regular", size: 16))
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:55]: The same

File: SenacorOnlineBanking/Components/SCInputField/SCInputField.swift [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+                    .opacity(viewModel.isContentVisible ? 1 : 0)
+                SecureField(viewModel.hint?.localized ?? "", text: $viewModel.value)
+                    .focused($focused, equals: .secure)
+                    .font(Font.custom("Roboto-Regular", size: 16))
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputField.swift:59]: The same

File: SenacorOnlineBanking/Components/SecureInfoView.swift [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+        HStack {
+            Image(.iconLock)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(width: 20, height: 20)
+            
+            Spacer().frame(width: 24)
+            
+            Text(.secureMessage)
+                .font(Font.custom("Roboto-Regular", size: 14))
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SecureInfoView.swift:21]: The same

File: SenacorOnlineBanking/Extensions/SwiftUI/ColorExtensions.swift [hunk 15]
```diff
@@ -0,0 +1,14 @@
+//
+//  ColorExtensions.swift
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Extensions/SwiftUI/ColorExtensions.swift:2]: Can we use Color+Extensions.swift?

File: SenacorOnlineBanking/Extensions/SwiftUI/TextExtensions.swift [hunk 16]
```diff
@@ -0,0 +1,14 @@
+//
+//  TextExtensions.swift
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Extensions/SwiftUI/TextExtensions.swift:2]: Can we use Text+Extensions.swift?

File: SenacorOnlineBanking/Extensions/SwiftUI/ViewExtensions.swift [hunk 17]
```diff
@@ -1,5 +1,5 @@
 //
-//  ViewExtension.swift
+//  ViewExtensions.swift
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Extensions/SwiftUI/ViewExtensions.swift:2]: Can we use View+Extensions.swift

File: SenacorOnlineBanking/Resources/de.lproj/Localizable.strings [hunk 18]
```diff
@@ -6,9 +6,14 @@
   
 */
 
+// common
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Resources/de.lproj/Localizable.strings:9]: Common?

File: SenacorOnlineBanking/Screens/Login/LoginView.swift [hunk 19]
```diff
@@ -12,30 +12,29 @@ struct LoginView: View {
...
+            VStack(spacing: 0) {
+                Text(viewModel.message)
+                    .font(Font.custom("Roboto-Bold", size: 14))
+                    .fgColor(.white)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+                    .padding(.all, 8)
+                    .bgColor(.primary)
+                    .cornerRadius(6)
+                
+                Spacer().frame(height: 8)
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginView.swift:24]: Please use padding instead

File: SenacorOnlineBanking/Screens/Login/LoginView.swift [hunk 20]
```diff
@@ -12,30 +12,29 @@ struct LoginView: View {
...
+                SCInputField(viewModel: viewModel.passwordViewModel)
                 
-                Button(action: viewModel.signInAction) {
-                    Text(viewModel.signInTitle)
-                        .fgColor(viewModel.signInForeground)
-                        .frame(maxWidth: .infinity)
-                        .padding(.vertical, 8)
-                        .bgColor(viewModel.signInBackground)
-                }
+                Spacer().frame(height: 32)
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginView.swift:29]: Please use padding instead

File: SenacorOnlineBanking/Screens/Login/LoginView.swift [hunk 21]
```diff
@@ -12,30 +12,29 @@ struct LoginView: View {
...
+                
+                SCButton(viewModel: viewModel.signInViewModel)
+                
+                Spacer().frame(height: 24)
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginView.swift:33]: Please use padding instead

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 22]
```diff
@@ -5,38 +5,57 @@
...
 }
 
 final class LoginViewModel: RendererViewModel, ObservableObject {
-    @Published var email: String = ""
-    @Published var password: String = ""
+    @Published var isValidDataProvided: Bool = false
+    private var cancellables = Set<AnyCancellable>()
     weak var coordinatorDelegate: LoginViewModelCoordinatorDelegate?
     
     var logoIcon: ImageAssets { .iconLogo }
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:20]: please use `let`

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 23]
```diff
@@ -5,38 +5,57 @@
...
-    var emailTitle: String { LocalizedStrings.emailTitle.localized }
-    var passwordTitle: String { LocalizedStrings.passwordTitle.localized }
-    var signInTitle: String { LocalizedStrings.signInTitle.localized }
+    var message: LocalizedStrings { .loginMessage }
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:21]: please use `let`

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 24]
```diff
@@ -5,38 +5,57 @@
...
     
-    var isSignInButtonActive: Bool {
-        email.isValidEmail && !password.isEmpty
-    }
+    lazy var emailViewModel: SCInputFieldViewModel = {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:23]: Please remove unnecessary closure
`lazy var emailViewModel: SCInputFieldViewModel = SCInputFieldViewModel(title: .emailTitle, isSecure: false)`

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 25]
```diff
@@ -5,38 +5,57 @@
...
+        SCInputFieldViewModel(title: .emailTitle, isSecure: false)
+    }()
     
-    var signInForeground: ColorAssets {
-        isSignInButtonActive ? .onAction : .onDisabled
-    }
+    lazy var passwordViewModel: SCInputFieldViewModel = {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:27]: Please remove unnecessary closure

File: SenacorOnlineBanking/Screens/Login/LoginViewModel.swift [hunk 26]
```diff
@@ -5,38 +5,57 @@
...
+        SCInputFieldViewModel(title: .passwordTitle, isSecure: true)
+    }()
     
-    var signInBackground: ColorAssets {
-        isSignInButtonActive ? .action : .disabled
+    var signInViewModel: SCButtonViewModel {
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Screens/Login/LoginViewModel.swift:31]: please use let

File: SenacorOnlineBanking/Components/SCInputField/SCInputFieldViewModel.swift [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+
+final class SCInputFieldViewModel: ObservableObject {
+    @Published var value: String = ""
+    @Published var isContentVisible: Bool = false
+    @Published var isFocused: Bool = false
+    let title: LocalizedStrings
+    let hint: LocalizedStrings?
+    let isSecure: Bool
+    
+    var secureIcon: ImageAssets { .iconEye }
```
Comment (pawel-finanteq) [inline SenacorOnlineBanking/Components/SCInputField/SCInputFieldViewModel.swift:18]: Please use `let`

**Review by pawel-finanteq — CHANGES_REQUESTED** [review]

File: SenacorOnlineBanking/Components/SCButton/SCButton.swift (hunk 1 above)
Comment (OlehPiskorskyj) [reply SenacorOnlineBanking/Components/SCButton/SCButton.swift:30]: It is more readable from my point of view. With this approach we will have only **var body: some View** in the main struct implementation scope, other subviews will be in extension.

**Review by pawel-finanteq — APPROVED** [review]

**Pull request merged**